### ISSUE TITLE
ExportTilesetDefinitionLinux.sh not being run correctly?

### ISSUE THREAD

**Issue by sommerper** [Cammin/LDtkUnity#58]
Hi!

I'm trying to get the plugin to work on Linux (PopOS) and it seems the command was added correctly to the ldtk file. The command inside the .sh file also points towards a working mono command.

![image](https://github.com/user-attachments/assets/c4813d2a-dcb2-4b7c-a181-55f0518c2a3b)

I've restarted both Unity and ldtk but I still get this error. Is there something else I could try to make it work?

**Comment by Cammin** [comment]
I can't personally test this OS so it's tricky for me. But, I'm willing to help hands-on in the discord server. 

I've had other Mac/Linux users help me with their issues, but maybe there's something else to clear up here 🙂

**Comment by sommerper** [comment]
I'll see if I have time for Discord at some point =)

But as far as I can see in `LDtkEditorCommandUpdater.cs`

You want `$1` to be a part of the command on mac an linux
https://github.com/Cammin/LDtkToUnity/blob/885860ca2db2d9d36870631861ea9cdaa8763d28/Assets/LDtkUnity/Editor/CustomEditor/Importer/LDtkEditorCommandUpdater.cs#L29

like so
```
../../../Library/LDtkTilesetExporter/ExportTilesetDefinitionLinux.sh "test_level" $1
```

but later you check if the split.length is not 2
https://github.com/Cammin/LDtkToUnity/blob/885860ca2db2d9d36870631861ea9cdaa8763d28/Assets/LDtkUnity/Editor/CustomEditor/Importer/LDtkEditorCommandUpdater.cs#L296

Which it is not because with the added `$1` it would be 3

So if I change the check to 3 it works.

But after some testing it seems like it really doesn't matter if the custom command in ldtk
is ```../../../Library/LDtkTilesetExporter/ExportTilesetDefinitionLinux.sh "test_level" $1``` or ```../../../Library/LDtkTilesetExporter/ExportTilesetDefinitionLinux.sh "test_level"```

it works either way and I can drag the level to the scene.

Thanks for the quick reply!

**Comment by Cammin** [comment]
Yeah, it is possible that it might be improperly thinking there is a problem when there is not, and so it shows that error dialog

**Comment by sommerper** [comment]
I could make some changes (i.e. pull request if you're accepting those) so that it works better on linux and mac.

I would also like to rename the .sh file to `../../../Library/LDtkTilesetExporter/ExportTilesetDefinition.sh` - omitting the Mac/Linux part so that you can work on both systems at the same time.

Right now I'm switching between Linux and Mac and I have to update the path in ldtk every time which is not super convenient.

**Comment by Haelle** [comment]
I'm using LDtk2Unity and it's working fine, the "fix" button generate the right command but the warning stays, I didn't figured out why thanks for the investigation.

cf https://github.com/Cammin/LDtkToUnity/pull/59 it wasn't working before 6.4.6

**Comment by Cammin** [comment]
It's definitely a string comparison issue, it thinks there is a problem when it shouldn't be
